packages/utils/src/testUtils.ts

```diff
@@ -53,6 +44,82 @@ export async function testEditor(spec: TestEditorSpec): Promise<void> {
     wrapper.remove();
 }
 
+/**
+ * Return a description of a range from analysing the position of
+ * `RANGE_TAIL_CHAR` and `RANGE_HEAD_CHAR` characters within a test container.
+ * Also remove these from the test container.
+ */
+function _parseTextualRange(testContainer: Node): DomRangeDescription {
+    let startContainer: Node;
+    let endContainer: Node;
+    let startOffset: number;
+    let endOffset: number;
+    let direction = Direction.FORWARD;
+    function __analyseChildren(node: Node): boolean {
+        return Array.from(node.childNodes).some(child => {
+            if (child.nodeType !== Node.TEXT_NODE) {
+                return child.hasChildNodes() && __analyseChildren(child);
+            }
+            const hasOnlyRange = Array.from(child.textContent).every(char => {
+                return char === RANGE_HEAD_CHAR || char === RANGE_TAIL_CHAR;
+            });
```

Comment:
When I thought about looking for the range characters in the DOM, I had the following algorithm in mind. Why wouldn't this work?
```
let sc, so, ec, eo;
node = root;
while (node && !(sc && so && ec && eo)) {
    if (node is a text node && node contains range char) {
        set sc so / ec ec 
    }
    node = next text node
} 
return sc, so, ec, eo
```